fdmi/plugins/m0scheduler

```diff
@@ -17,128 +17,277 @@
+
+    # if record is not dedupable, add to cache
+    if flag and kv['cr_val'] != "0":
+        print('new kv: {}'.format(kv), file=sys.stderr)
+        kv_records.append(kv)
+
+
+# handle SIGNT and SIGTERM
 def signal_handler(sig, frame):
-	print("CTRL + C pressed")
-	process.send_signal(sig)
-	process.wait()
-	#print(kv_records)
-	sys.exit(0)
+    print('CTRL + C pressed', file=sys.stderr)
+    print('Stopping all m0sched processes...')
+    for p in processes:
+        p.send_signal(sig)
+        p.wait()
+
+    # print(kv_records)
+    sys.exit(0)
+
 
 def execute_command_and_get_output(cmd):
     proc = subprocess.Popen(cmd, stdin=subprocess.PIPE,
-                                    stdout=subprocess.PIPE,
-                                    stderr=subprocess.PIPE,
-                                    shell=True)
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            shell=True)
     stdout, stderr = proc.communicate()
     output_var = str(stdout, 'utf-8')
     return output_var
 
 
-#m0sched -l 192.168.52.53@tcp:12345:4:1 -h 192.168.52.53@tcp:12345:1:1 -p 0x7000000000000001:0x37 -f 0x7200000000000001:0x19
-def get_m0sched_cmd():
-	#m0sched = "/home/931442/cortx-motr/fdmi/plugins/m0sched"
-	m0sched = "m0sched"
-	opts = " -l " + local_endpoint + " -h " + ha_endpoint + " -p " + profile_fid + " -f " + process_fid + " -g " + fdmi_filter_fid
-	return m0sched + opts
-
-#get local ep, ha ep, profile fid and process fid
-def get_cluster_info():
-    ret = 0
-    global profile_fid
-    global ha_endpoint
-    global local_endpoint
-    global process_fid
-    global fdmi_filter_fid
-    cmd = 'hctl status --json'
-    cluster_info = execute_command_and_get_output(cmd)
-    cluster_js = json.loads(cluster_info)
-    if (cluster_info != None):
-        profile_fid = str(cluster_js["profiles"][0]["fid"])
-        nodes_data = cluster_js["nodes"]
-        for svcs in nodes_data[0]['svcs']:
-            if svcs['name'] == 'hax':
-                ha_endpoint = str(svcs['ep'])
-
-            #m0sched alway uses first m0client
-            if svcs['name'] == 'm0_client':
-                local_endpoint = str(svcs['ep'])
-                process_fid = str(svcs['fid'])
-                break
+# m0sched -l 192.168.52.53@tcp:12345:4:1 -h 192.168.52.53@tcp:12345:1:1 -p 0x7000000000000001:0x37
+# -f 0x7200000000000001:0x19
+def get_scheduler_cmd(args):
+    # m0sched = "/home/931442/cortx-motr/fdmi/plugins/m0sched"
+    m0sched = args.scheduler_path
+    opts = " -l " + cluster_info['local_endpoint'] + " -h " + cluster_info['ha_endpoint'] + \
+           " -p " + cluster_info['profile_fid'] + " -f " + cluster_info['process_fid'] + " -g " + \
+           cluster_info['fdmi_filter_fid']
+    return m0sched + opts
+
+
+# get local ep, ha ep, profile fid and process fid
+def get_cluster_info(args):
+    info = dict()
+    cmd = '{} status --json'.format(args.hctl_path)
```

Comment:
In python3, we can use "The Python Formatted String Literal (f-String)".
```
cmd = f'{args.hctl_path} status --json'
```

But this is not mandatory. A developer can choose his favorite way.

Reply:
Well, to me it looks less obvious than the way I used it ;)

Reply:
OK